src/libs/elektra/opmphm_vheap.c

```diff
@@ -0,0 +1,141 @@
+#include "kdbprivate.h"
+
+/**
+ * Allocates vheap with size specified by parameter minSize
+ *
+ * To construct a max heap the comparison function VheapComp (a, b), must return
+ * 1 on a > b and 0 otherwise. For a min heap 1 on a < b and 0 otherwise.
+ *
+ * @ingroup vheap
+ * @param comp the comparison function of the heap
+ * @param minSize the minimum size of the heap
+ * @return a Vheap pointer
+ * @return NULL error
+ */
+Vheap * elektraVheapInit (VheapComp comp, size_t minSize)
+{
+	if (minSize < 1 || !comp) return NULL;
+	Vheap * newVheap = elektraMalloc (sizeof (Vheap));
+	if (!newVheap) return NULL;
+	newVheap->comp = comp;
+	newVheap->size = minSize;
+	newVheap->minSize = minSize;
+	newVheap->count = 0;
+	newVheap->data = elektraMalloc (newVheap->size * sizeof (void *));
+	if (!newVheap->data)
+	{
+		elektraFree (newVheap);
+		return NULL;
+	}
+	return newVheap;
+}
+
+/**
+ * Checks if the heap is empty.
+ *
+ * @ingroup vheap
+ * @return 1 on empty
+ * @return 0 on non empty
+ */
+int elektraVheapIsEmpty (const Vheap * vheap)
+{
+	if (!vheap) return 0;
+	return (vheap->count == 0) ? 1 : 0;
```

Comment:
I think it would be even easier if we just use `!vheap->count` instead of `(vheap->count == 0) ? 1 : 0` here.

Overall I would replace the two lines with something like this:

```C
// A heap is empty if it exists and its size is 0
return vheap && vheap->count == 0;
```

In theory it should just work as well. At least `test_opmphm_vheap` still works after I change the code. Please keep in mind that this is just a suggestion :o). If you like the code as it is now, there is no need to change it.